grants_tagger/train_with_sagemaker.py

```diff
@@ -0,0 +1,132 @@
+import tarfile
+import logging
+import json
+import os
+
+from sagemaker.estimator import Estimator
+import sagemaker
+
+
+logger = logging.getLogger(__name__)
+
+ENTRYPOINT = "grants_tagger/train.py"
+DEPENDENCIES = [
+    "grants_tagger/__init__.py",
+    "grants_tagger/utils.py",
+    "grants_tagger/models.py"
+]
+try:
+    PROJECTS_BUCKET = os.environ["PROJECTS_BUCKET"]
+    PROJECT_NAME = os.environ["PROJECT_NAME"]
+    SAGEMAKER_ROLE = os.environ["SAGEMAKER_ROLE"]
+except KeyError:
+    logger.warning("One or more of PROJECTS_BUCKET, PROJECT_NAME, SAGEMAKER_ROLE is not set. SageMaker will not work.")
+
+def upload_code(entrypoint, dependencies, bucket, prefix):
+    """
+    Upload code for train
+
+    Args:
+    entrypoint: path to train script
+    dependencies: list of path to additional source files needed
+    bucket: s3 bucket that sagemaker will use to read and write artifacts
+    prefix: s3 prefix     -//- (same as above)
+    """
+    session = sagemaker.Session()
+    
+    source_files = [entrypoint] + dependencies
+    logger.info(source_files)
+        
+    tmp_dir = "/tmp"
+    filename = os.path.join(tmp_dir, "sourcedir.tar.gz")
+    
+    with tarfile.open(filename, mode="w:gz") as t:
+        for sf in source_files:
+            t.add(sf)
+
+    # Probably can be replaced by boto
+    s3_path = session.upload_data(filename, bucket, prefix + '/code') 
```

Comment:
Are these folder names (`/tmp`, `code` etc) SageMaker specific or can you rename?

Reply:
`/tmp` is the default temp folder in unix systems

`code` can be anything but sagemaker uses the name code folder for inside the container so I kept it for outside the container too.